src/Core/Geographic/GeoCoordinate.js

```diff
@@ -18,60 +21,73 @@ export const UNIT = {
     METER: 2,
 };
 
-var getCoordinateValue = function getCoordinateValue(unit, coord, id)
-{
-    unit = unit || UNIT.RADIAN;
-
-    if (unit === UNIT.RADIAN) {
-        return coord[id];
-    } else if (unit === UNIT.DEGREE) {
-        return mE.radToDeg(coord[id]);
-    }
-};
-
-var setCoordinateValue = function setCoordinateValue(unit, coord, id, value)
-{
-    unit = unit || UNIT.RADIAN;
-
-    if (unit === UNIT.RADIAN) {
-        coord[id] = value;
-    } else if (unit === UNIT.DEGREE) {
-        coord[id] = mE.degToRad(value);
+// Only support explicit conversions
+function convert(inUnit, outUnit, value) {
+    if (inUnit == outUnit || outUnit === undefined) {
+        return value;
+    } else {
+        switch (inUnit) {
+            case UNIT.RADIAN: {
+                if (outUnit === UNIT.DEGREE) {
+                    return mE.radToDeg(value);
+                } else {
+                    // from meter require CRS and projection etc
+                    throw new Error('Cannot convert from Meters to Radians');
+                }
+            }
+            case UNIT.DEGREE: {
+                if (outUnit === UNIT.RADIAN) {
+                    return mE.degToRad(value);
+                } else {
+                    // from meter require CRS and projection etc
+                    throw new Error('Cannot convert from Meters to Degree');
+                }
+            }
+            default:
+                throw new Error(`Cannot convert from unit ${inUnit} to unit ${outUnit}`);
+        }
     }
-    return coord[id];
-};
-
-var setCoordinate = function setCoordinate(coordinate, longitude, latitude, altitude, unit) {
-    unit = unit || UNIT.RADIAN;
-
-    setCoordinateValue(unit, coordinate, COORD.LONG, longitude);
-    setCoordinateValue(unit, coordinate, COORD.LAT, latitude);
+}
 
+var setCoordinate = function setCoordinate(coordinate, longitude, latitude, altitude, inUnit, toUnit) {
+    coordinate[COORD.LONG] = convert(inUnit, toUnit, longitude);
+    coordinate[COORD.LAT] = convert(inUnit, toUnit, latitude);
     coordinate[COORD.ALT] = altitude;
 };
 
 function GeoCoordinate(longitude, latitude, altitude, unit) {
+    if (longitude && unit === undefined) {
+        throw new Error('Cannot build a GeoCoordinate without a unit');
+    }
+
+    this.unit = unit;
     this.coordinate = new Float64Array(3);
 
-    setCoordinate(this.coordinate, longitude, latitude, altitude, unit);
+    setCoordinate(this.coordinate, longitude, latitude, altitude, unit, unit);
 }
 
 GeoCoordinate.prototype.constructor = GeoCoordinate;
 
 GeoCoordinate.prototype.longitude = function longitude(unit) {
-    return getCoordinateValue(unit, this.coordinate, COORD.LONG);
+    return convert(this.unit, unit, this.coordinate[COORD.LONG]);
 };
 
+GeoCoordinate.prototype.x = GeoCoordinate.prototype.longitude;
```

Comment:
Coordinates will be at the same time a cartesian object and a spherical object, it is not common in terms of object. Generally the basic objects are separated, for example :

**THREE**

[`THREE.Vector3`](https://threejs.org/docs/?q=vect#Reference/Math/Vector3) and [`THREE.Spherical`](https://threejs.org/docs/?q=spher#Reference/Math/Spherical) 
 
and some specific methods to set object from another : 
`THREE.Vector3.setFromSpherical(sperical)`
`THREE.Spherical.setFromVector3(vector3)`

**Cesium**

[`Cesium.Cartesian3`](https://cesiumjs.org/Cesium/Build/Documentation/Cartesian3.html) and 
[`Cesium.Cartographic`](https://cesiumjs.org/Cesium/Build/Documentation/Cartographic.html)

Methods

`Cesium.Cartesian3.fromCartesian(cartesian, ellipsoid, result)`
`Cesium.Cartesian3.fromRadians(longitude, latitude, height, ellipsoid, result)`


There is `Projection` object  (Which is a bit in _iTowns_) in Cesium to convert object 

[GeographicProjection](https://cesiumjs.org/Cesium/Build/Documentation/GeographicProjection.html)